File: test_api_markdown.py
#!/usr/bin/env python3
"""
Test script for Gnosis Wraith /api/markdown endpoint
Tests both single URL and batch URL processing
Based on the actual implementation in web/routes/apis/api.py
"""

import asyncio
import aiohttp
import json
import time
from typing import List, Dict, Any

# Configuration
REMOTE_SERVER = "https://wraith.nuts.services"
API_KEY = "NI1qoVZLEgTVxWFr7lxdx_10ZvqGRgYEEEPa9uS1WHwjkdUkEqh72Q"  # Your API key

# Test URLs
TEST_URLS = [
    "https://example.com",
    "https://www.wikipedia.org",
    "https://news.ycombinator.com",
    "https://www.python.org",
    "https://docs.python.org/3/"
]

async def test_single_url_markdown():
    """Test single URL markdown extraction (backward compatible)"""
    print("\n=== Testing Single URL Markdown ===")
    
    headers = {"Authorization": f"Bearer {API_KEY}"}
    payload = {
        "url": TEST_URLS[0],
        "javascript_enabled": True,
        "screenshot_mode": None  # No screenshot for speed
    }
    
    async with aiohttp.ClientSession() as session:
        try:
            start_time = time.time()
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                elapsed = time.time() - start_time
                
                print(f"URL: {TEST_URLS[0]}")
                print(f"Status: {response.status}")
                print(f"Time: {elapsed:.2f}s")
                
                if response.status == 200:
                    data = await response.json()
                    print(f"✅ Success")
                    print(f"Response keys: {list(data.keys())}")
                    
                    # Check various response fields
                    if 'success' in data:
                        print(f"Success: {data['success']}")
                    if 'url' in data:
                        print(f"URL: {data['url']}")
                    if 'markdown_url' in data:
                        print(f"Markdown URL: {data['markdown_url']}")
                    if 'json_url' in data:
                        print(f"JSON URL: {data['json_url']}")
                    if 'stats' in data:
                        print(f"Stats: {data['stats']}")
                else:
                    error_data = await response.json()
                    print(f"❌ Error: {error_data}")
                    
                return response.status == 200
                
        except Exception as e:
            print(f"❌ Exception: {e}")
            return False

async def test_batch_sync():
    """Test synchronous batch processing"""
    print("\n=== Testing Batch Markdown (Sync) ===")
    
    headers = {"Authorization": f"Bearer {API_KEY}"}
    payload = {
        "urls": TEST_URLS[:3],  # Test with 3 URLs
        "async": False,  # Wait for completion
        "javascript_enabled": True,
        "screenshot_mode": None
    }
    
    async with aiohttp.ClientSession() as session:
        try:
            start_time = time.time()
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                elapsed = time.time() - start_time
                
                print(f"URLs: {len(payload['urls'])} URLs")
                print(f"Status: {response.status}")
                print(f"Time: {elapsed:.2f}s")
                
                if response.status == 200:
                    data = await response.json()
                    print(f"✅ Success")
                    print(f"Mode: {data.get('mode', 'N/A')}")
                    print(f"Job ID: {data.get('job_id', 'N/A')}")
                    
                    if 'results' in data:
                        print(f"\nResults: {len(data['results'])} items")
                        for i, result in enumerate(data['results'][:2]):  # Show first 2
                            print(f"\nResult {i+1}:")
                            print(f"  URL: {result.get('url', 'N/A')}")
                            print(f"  Status: {result.get('status', 'N/A')}")
                            print(f"  Markdown URL: {result.get('markdown_url', 'N/A')}")
                            print(f"  JSON URL: {result.get('json_url', 'N/A')}")
                else:
                    error_data = await response.json()
                    print(f"❌ Error: {error_data}")
                    
                return response.status == 200
                
        except Exception as e:
            print(f"❌ Exception: {e}")
            return False

async def test_batch_async():
    """Test asynchronous batch processing"""
    print("\n=== Testing Batch Markdown (Async) ===")
    
    headers = {"Authorization": f"Bearer {API_KEY}"}
    payload = {
        "urls": TEST_URLS[:4],  # Test with 4 URLs
        "async": True,  # Don't wait for completion
        "javascript_enabled": False,  # Faster without JS
        "screenshot_mode": None
    }
    
    async with aiohttp.ClientSession() as session:
        try:
            start_time = time.time()
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                elapsed = time.time() - start_time
                
                print(f"URLs: {len(payload['urls'])} URLs")
                print(f"Status: {response.status}")
                print(f"Time: {elapsed:.2f}s (should be fast for async)")
                
                if response.status in [200, 202]:  # 202 Accepted for async
                    data = await response.json()
                    print(f"✅ Success (Status: {response.status})")
                    print(f"Mode: {data.get('mode', 'N/A')}")
                    print(f"Job ID: {data.get('job_id', 'N/A')}")
                    print(f"Status URL: {data.get('status_url', 'N/A')}")
                    
                    if 'results' in data:
                        print(f"\nPredicted URLs: {len(data['results'])} items")
                        for i, result in enumerate(data['results'][:2]):
                            print(f"  {i+1}. {result.get('url', 'N/A')} - Status: {result.get('status', 'N/A')}")
                else:
                    error_data = await response.json()
                    print(f"❌ Error: {error_data}")
                    
                return response.status in [200, 202]
                
        except Exception as e:
            print(f"❌ Exception: {e}")
            return False

async def test_batch_with_collation():
    """Test batch processing with collation"""
    print("\n=== Testing Batch with Collation ===")
    
    headers = {"Authorization": f"Bearer {API_KEY}"}
    payload = {
        "urls": [
            "https://docs.python.org/3/tutorial/introduction.html",
            "https://docs.python.org/3/tutorial/controlflow.html"
        ],
        "async": False,  # Wait for completion to get collated result
        "collate": True,
        "collate_options": {
            "title": "Python Tutorial Collection",
            "add_toc": True,
            "add_source_headers": True
        },
        "javascript_enabled": False
    }
    
    async with aiohttp.ClientSession() as session:
        try:
            start_time = time.time()
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                elapsed = time.time() - start_time
                
                print(f"URLs: {len(payload['urls'])} URLs")
                print(f"Status: {response.status}")
                print(f"Time: {elapsed:.2f}s")
                
                if response.status == 200:
                    data = await response.json()
                    print(f"✅ Success")
                    
                    if 'collated_url' in data:
                        print(f"Collated file URL: {data['collated_url']}")
                    
                    if 'results' in data:
                        print(f"Individual results: {len(data['results'])}")
                else:
                    error_data = await response.json()
                    print(f"❌ Error: {error_data}")
                    
                return response.status == 200
                
        except Exception as e:
            print(f"❌ Exception: {e}")
            return False

async def test_with_filters():
    """Test markdown extraction with content filters"""
    print("\n=== Testing with Content Filters ===")
    
    headers = {"Authorization": f"Bearer {API_KEY}"}
    
    # Test with pruning filter
    payload = {
        "url": "https://en.wikipedia.org/wiki/Python_(programming_language)",
        "filter": "pruning",
        "filter_options": {
            "threshold": 0.5,
            "min_words": 3
        },
        "javascript_enabled": False
    }
    
    async with aiohttp.ClientSession() as session:
        try:
            print("Testing Pruning Filter...")
            start_time = time.time()
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                elapsed = time.time() - start_time
                
                if response.status == 200:
                    data = await response.json()
                    print(f"  ✅ Pruning filter success (Time: {elapsed:.2f}s)")
                    if 'stats' in data:
                        print(f"  Stats: {data['stats']}")
                else:
                    print(f"  ❌ Pruning filter failed: Status {response.status}")
            
            # Test with BM25 filter
            payload["filter"] = "bm25"
            payload["filter_options"] = {
                "query": "python programming syntax",
                "threshold": 0.4
            }
            
            print("\nTesting BM25 Filter...")
            start_time = time.time()
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                elapsed = time.time() - start_time
                
                if response.status == 200:
                    data = await response.json()
                    print(f"  ✅ BM25 filter success (Time: {elapsed:.2f}s)")
                    if 'stats' in data:
                        print(f"  Stats: {data['stats']}")
                else:
                    print(f"  ❌ BM25 filter failed: Status {response.status}")
                    
            return True
                    
        except Exception as e:
            print(f"❌ Exception: {e}")
            return False

async def test_error_handling():
    """Test error handling with invalid URLs"""
    print("\n=== Testing Error Handling ===")
    
    headers = {"Authorization": f"Bearer {API_KEY}"}
    
    # Test invalid URL
    payload = {
        "url": "not-a-valid-url"
    }
    
    async with aiohttp.ClientSession() as session:
        try:
            print("Testing invalid URL...")
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                data = await response.json()
                print(f"  Status: {response.status}")
                print(f"  Response: {data}")
                
            # Test missing URL
            payload = {}
            print("\nTesting missing URL...")
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                data = await response.json()
                print(f"  Status: {response.status}")
                print(f"  Response: {data}")
                
            # Test too many URLs
            payload = {
                "urls": ["https://example.com"] * 51  # Over 50 limit
            }
            print("\nTesting too many URLs...")
            async with session.post(f"{REMOTE_SERVER}/api/markdown", json=payload, headers=headers) as response:
                data = await response.json()
                print(f"  Status: {response.status}")
                print(f"  Response: {data}")
                
            return True
                
        except Exception as e:
            print(f"❌ Exception: {e}")
            return False

async def main():
    """Run all tests"""
    print("=" * 60)
    print("Gnosis Wraith /api/markdown Endpoint Tests")
    print(f"Server: {REMOTE_SERVER}")
    print(f"API Key: {API_KEY[:10]}...")
    print("=" * 60)
    
    # Run tests
    tests = [
        ("Single URL", test_single_url_markdown),
        ("Batch Sync", test_batch_sync),
        ("Batch Async", test_batch_async),
        ("Batch with Collation", test_batch_with_collation),
        ("Content Filters", test_with_filters),
        ("Error Handling", test_error_handling),
    ]
    
    results = []
    for test_name, test_func in tests:
        try:
            result = await test_func()
            results.append((test_name, result))
        except Exception as e:
            print(f"\n❌ Test '{test_name}' crashed: {e}")
            results.append((test_name, False))
    
    # Summary
    print("\n" + "=" * 60)
    print("Test Summary")
    print("=" * 60)
    
    passed = sum(1 for _, result in results if result)
    total = len(results)
    
    for test_name, result in results:
        status = "✅ PASSED" if result else "❌ FAILED"
        print(f"{test_name}: {status}")
    
    print(f"\nTotal: {passed}/{total} tests passed")
    
    print("\n📝 Implementation Notes:")
    print("- The /api/markdown endpoint supports both single and batch URLs")
    print("- Batch mode is triggered by providing 'urls' array instead of 'url'")
    print("- Maximum 50 URLs per batch")
    print("- Supports async processing with webhooks")
    print("- Supports content filters (pruning, BM25)")
    print("- Supports collation to merge results")

if __name__ == "__main__":
    asyncio.run(main())

File: test_api_markdown.js
#!/usr/bin/env node
/**
 * Test script for Gnosis Wraith /api/markdown endpoint
 * Tests both single URL and batch URL processing
 * Node.js equivalent of test_api_markdown.py
 */

const https = require('https');
const http = require('http');

// Configuration
const REMOTE_SERVER = "https://wraith.nuts.services";
// const REMOTE_SERVER = "http://localhost:5678";
const API_KEY = "Ij8zhKKL_iKUcumU5oQKnpsECg9qvYTdXlH2IGth7k_ewGBdy6se_g"; // Your API key
// const API_KEY = "pfr3lKBeacc4W5ZiOA6jFYRFvywhd4_MoHg8MkFmeesSvdmPMNheA";

// Test URLs
const TEST_URLS = [
    "https://example.com",
    "https://www.wikipedia.org",
    "https://news.ycombinator.com",
    "https://www.python.org",
    "https://docs.python.org/3/"
];

/**
 * Make HTTP request helper
 */
function makeRequest(url, options, payload) {
    return new Promise((resolve, reject) => {
        const urlObj = new URL(url);
        const protocol = urlObj.protocol === 'https:' ? https : http;
        
        const reqOptions = {
            hostname: urlObj.hostname,
            port: urlObj.port || (urlObj.protocol === 'https:' ? 443 : 80),
            path: urlObj.pathname + urlObj.search,
            method: options.method || 'GET',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${API_KEY}`,
                ...options.headers
            }
        };

        const req = protocol.request(reqOptions, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => {
                try {
                    const jsonData = JSON.parse(data);
                    resolve({ status: res.statusCode, data: jsonData });
                } catch (e) {
                    resolve({ status: res.statusCode, data: data });
                }
            });
        });

        req.on('error', reject);
        
        if (payload) {
            req.write(JSON.stringify(payload));
        }
        
        req.end();
    });
}

/**
 * Test single URL markdown extraction (backward compatible)
 */
async function testSingleUrlMarkdown() {
    console.log("\n=== Testing Single URL Markdown ===");
    
    const payload = {
        url: TEST_URLS[0],
        javascript_enabled: true,
        screenshot_mode: null // No screenshot for speed
    };
    
    try {
        const startTime = Date.now();
        const response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            payload
        );
        const elapsed = (Date.now() - startTime) / 1000;
        
        console.log(`URL: ${TEST_URLS[0]}`);
        console.log(`Status: ${response.status}`);
        console.log(`Time: ${elapsed.toFixed(2)}s`);
        
        if (response.status === 200) {
            const data = response.data;
            console.log("✅ Success");
            console.log(`Response keys: ${Object.keys(data).join(', ')}`);
            
            // Check various response fields
            if ('success' in data) {
                console.log(`Success: ${data.success}`);
            }
            if ('url' in data) {
                console.log(`URL: ${data.url}`);
            }
            if ('markdown_url' in data) {
                console.log(`Markdown URL: ${data.markdown_url}`);
            }
            if ('json_url' in data) {
                console.log(`JSON URL: ${data.json_url}`);
            }
            if ('stats' in data) {
                console.log(`Stats:`, data.stats);
            }
        } else {
            console.log(`❌ Error:`, response.data);
        }
        
        return response.status === 200;
        
    } catch (error) {
        console.log(`❌ Exception: ${error.message}`);
        return false;
    }
}

/**
 * Test synchronous batch processing
 */
async function testBatchSync() {
    console.log("\n=== Testing Batch Markdown (Sync) ===");
    
    const payload = {
        urls: TEST_URLS.slice(0, 3), // Test with 3 URLs
        async: false, // Wait for completion
        javascript_enabled: true,
        screenshot_mode: null
    };
    
    try {
        const startTime = Date.now();
        const response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            payload
        );
        const elapsed = (Date.now() - startTime) / 1000;
        
        console.log(`URLs: ${payload.urls.length} URLs`);
        console.log(`Status: ${response.status}`);
        console.log(`Time: ${elapsed.toFixed(2)}s`);
        
        if (response.status === 200) {
            const data = response.data;
            console.log("✅ Success");
            console.log(`Mode: ${data.mode || 'N/A'}`);
            console.log(`Job ID: ${data.job_id || 'N/A'}`);
            
            if ('results' in data) {
                console.log(`\nResults: ${data.results.length} items`);
                data.results.slice(0, 2).forEach((result, i) => {
                    console.log(`\nResult ${i + 1}:`);
                    console.log(`  URL: ${result.url || 'N/A'}`);
                    console.log(`  Status: ${result.status || 'N/A'}`);
                    console.log(`  Markdown URL: ${result.markdown_url || 'N/A'}`);
                    console.log(`  JSON URL: ${result.json_url || 'N/A'}`);
                });
            }
        } else {
            console.log(`❌ Error:`, response.data);
        }
        
        return response.status === 200;
        
    } catch (error) {
        console.log(`❌ Exception: ${error.message}`);
        return false;
    }
}

/**
 * Test asynchronous batch processing
 */
async function testBatchAsync() {
    console.log("\n=== Testing Batch Markdown (Async) ===");
    
    const payload = {
        urls: TEST_URLS.slice(0, 4), // Test with 4 URLs
        async: true, // Don't wait for completion
        javascript_enabled: false, // Faster without JS
        screenshot_mode: null
    };
    
    try {
        const startTime = Date.now();
        const response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            payload
        );
        const elapsed = (Date.now() - startTime) / 1000;
        
        console.log(`URLs: ${payload.urls.length} URLs`);
        console.log(`Status: ${response.status}`);
        console.log(`Time: ${elapsed.toFixed(2)}s (should be fast for async)`);
        
        if ([200, 202].includes(response.status)) { // 202 Accepted for async
            const data = response.data;
            console.log(`✅ Success (Status: ${response.status})`);
            console.log(`Mode: ${data.mode || 'N/A'}`);
            console.log(`Job ID: ${data.job_id || 'N/A'}`);
            console.log(`Status URL: ${data.status_url || 'N/A'}`);
            
            if ('results' in data) {
                console.log(`\nPredicted URLs: ${data.results.length} items`);
                data.results.slice(0, 2).forEach((result, i) => {
                    console.log(`  ${i + 1}. ${result.url || 'N/A'} - Status: ${result.status || 'N/A'}`);
                });
            }
        } else {
            console.log(`❌ Error:`, response.data);
        }
        
        return [200, 202].includes(response.status);
        
    } catch (error) {
        console.log(`❌ Exception: ${error.message}`);
        return false;
    }
}

/**
 * Test batch processing with collation
 */
async function testBatchWithCollation() {
    console.log("\n=== Testing Batch with Collation ===");
    
    const payload = {
        urls: [
            "https://docs.python.org/3/tutorial/introduction.html",
            "https://docs.python.org/3/tutorial/controlflow.html"
        ],
        async: false, // Wait for completion to get collated result
        collate: true,
        collate_options: {
            title: "Python Tutorial Collection",
            add_toc: true,
            add_source_headers: true
        },
        javascript_enabled: false
    };
    
    try {
        const startTime = Date.now();
        const response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            payload
        );
        const elapsed = (Date.now() - startTime) / 1000;
        
        console.log(`URLs: ${payload.urls.length} URLs`);
        console.log(`Status: ${response.status}`);
        console.log(`Time: ${elapsed.toFixed(2)}s`);
        
        if (response.status === 200) {
            const data = response.data;
            console.log("✅ Success");
            
            if ('collated_url' in data) {
                console.log(`Collated file URL: ${data.collated_url}`);
            }
            
            if ('results' in data) {
                console.log(`Individual results: ${data.results.length}`);
            }
        } else {
            console.log(`❌ Error:`, response.data);
        }
        
        return response.status === 200;
        
    } catch (error) {
        console.log(`❌ Exception: ${error.message}`);
        return false;
    }
}

/**
 * Test markdown extraction with content filters
 */
async function testWithFilters() {
    console.log("\n=== Testing with Content Filters ===");
    
    try {
        // Test with pruning filter
        console.log("Testing Pruning Filter...");
        let payload = {
            url: "https://en.wikipedia.org/wiki/Python_(programming_language)",
            filter: "pruning",
            filter_options: {
                threshold: 0.5,
                min_words: 3
            },
            javascript_enabled: false
        };
        
        let startTime = Date.now();
        let response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            payload
        );
        let elapsed = (Date.now() - startTime) / 1000;
        
        if (response.status === 200) {
            const data = response.data;
            console.log(`  ✅ Pruning filter success (Time: ${elapsed.toFixed(2)}s)`);
            if ('stats' in data) {
                console.log(`  Stats:`, data.stats);
            }
        } else {
            console.log(`  ❌ Pruning filter failed: Status ${response.status}`);
        }
        
        // Test with BM25 filter
        console.log("\nTesting BM25 Filter...");
        payload = {
            url: "https://en.wikipedia.org/wiki/Python_(programming_language)",
            filter: "bm25",
            filter_options: {
                query: "python programming syntax",
                threshold: 0.4
            },
            javascript_enabled: false
        };
        
        startTime = Date.now();
        response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            payload
        );
        elapsed = (Date.now() - startTime) / 1000;
        
        if (response.status === 200) {
            const data = response.data;
            console.log(`  ✅ BM25 filter success (Time: ${elapsed.toFixed(2)}s)`);
            if ('stats' in data) {
                console.log(`  Stats:`, data.stats);
            }
        } else {
            console.log(`  ❌ BM25 filter failed: Status ${response.status}`);
        }
        
        return true;
        
    } catch (error) {
        console.log(`❌ Exception: ${error.message}`);
        return false;
    }
}

/**
 * Test error handling with invalid URLs
 */
async function testErrorHandling() {
    console.log("\n=== Testing Error Handling ===");
    
    try {
        // Test invalid URL
        console.log("Testing invalid URL...");
        let response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            { url: "not-a-valid-url" }
        );
        console.log(`  Status: ${response.status}`);
        console.log(`  Response:`, response.data);
        
        // Test missing URL
        console.log("\nTesting missing URL...");
        response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            {}
        );
        console.log(`  Status: ${response.status}`);
        console.log(`  Response:`, response.data);
        
        // Test too many URLs
        console.log("\nTesting too many URLs...");
        response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            { urls: Array(51).fill("https://example.com") } // Over 50 limit
        );
        console.log(`  Status: ${response.status}`);
        console.log(`  Response:`, response.data);
        
        return true;
        
    } catch (error) {
        console.log(`❌ Exception: ${error.message}`);
        return false;
    }
}

/**
 * Test minimal response format
 */
async function testMinimalResponseFormat() {
    console.log("\n=== Testing Minimal Response Format ===");
    
    const payload = {
        url: "https://example.com",
        javascript_enabled: false, // Faster without JS
        screenshot_mode: null,
        response_format: "minimal" // Request minimal format
    };
    
    try {
        const startTime = Date.now();
        const response = await makeRequest(
            `${REMOTE_SERVER}/api/markdown`,
            { method: 'POST' },
            payload
        );
        const elapsed = (Date.now() - startTime) / 1000;
        
        console.log(`URL: ${payload.url}`);
        console.log(`Response Format: ${payload.response_format}`);
        console.log(`Status: ${response.status}`);
        console.log(`Time: ${elapsed.toFixed(2)}s`);
        
        if (response.status === 200) {
            const data = response.data;
            console.log("✅ Success");
            console.log(`Response keys: ${Object.keys(data).join(', ')}`);
            
            // Output the content
            console.log("\n--- CONTENT OUTPUT ---");
            if ('content' in data) {
                console.log("Content field:");
                console.log(data.content ? data.content.substring(0, 500) + '...' : 'No content');
            }
            if ('markdown' in data) {
                console.log("\nMarkdown field:");
                console.log(data.markdown ? data.markdown.substring(0, 500) + '...' : 'No markdown');
            }
            console.log("--- END CONTENT ---\n");
            
            // Check what fields are included in minimal format
            console.log("\nFields included:");
            const fields = ['success', 'url', 'title', 'content', 'markdown', 'markdown_url', 'json_url', 'stats', 'html_content', 'extracted_text'];
            fields.forEach(field => {
                if (field in data) {
                    console.log(`  ✓ ${field}: ${typeof data[field] === 'string' && data[field].length > 50 ? data[field].substring(0, 50) + '...' : JSON.stringify(data[field])}`);
                } else {
                    console.log(`  ✗ ${field}: not included`);
                }
            });
            
        } else {
            console.log(`❌ Error:`, response.data);
        }
        
        return response.status === 200;
        
    } catch (error) {
        console.log(`❌ Exception: ${error.message}`);
        return false;
    }
}

/**
 * Main test runner
 */
async function main() {
    console.log("=".repeat(60));
    console.log("Gnosis Wraith /api/markdown Endpoint Tests");
    console.log(`Server: ${REMOTE_SERVER}`);
    console.log(`API Key: ${API_KEY.substring(0, 10)}...`);
    console.log("=".repeat(60));
    
    // Run tests
    const tests = [
        { name: "Minimal Response Format", func: testMinimalResponseFormat }, // Run this first
        { name: "Single URL", func: testSingleUrlMarkdown },
        { name: "Batch Sync", func: testBatchSync },
        { name: "Batch Async", func: testBatchAsync },
        { name: "Batch with Collation", func: testBatchWithCollation },
        { name: "Content Filters", func: testWithFilters },
        { name: "Error Handling", func: testErrorHandling }
    ];

    
    const results = [];
    for (const test of tests) {
        try {
            const result = await test.func();
            results.push({ name: test.name, passed: result });
        } catch (error) {
            console.log(`\n❌ Test '${test.name}' crashed: ${error.message}`);
            results.push({ name: test.name, passed: false });
        }
    }
    
    // Summary
    console.log("\n" + "=".repeat(60));
    console.log("Test Summary");
    console.log("=".repeat(60));
    
    const passed = results.filter(r => r.passed).length;
    const total = results.length;
    
    results.forEach(result => {
        const status = result.passed ? "✅ PASSED" : "❌ FAILED";
        console.log(`${result.name}: ${status}`);
    });
    
    console.log(`\nTotal: ${passed}/${total} tests passed`);
    
    console.log("\n📝 Implementation Notes:");
    console.log("- The /api/markdown endpoint supports both single and batch URLs");
    console.log("- Batch mode is triggered by providing 'urls' array instead of 'url'");
    console.log("- Maximum 50 URLs per batch");
    console.log("- Supports async processing with webhooks");
    console.log("- Supports content filters (pruning, BM25)");
    console.log("- Supports collation to merge results");
}

// Run the tests
main().catch(console.error);